src/gas-price/gas-price.ts

```diff
@@ -10,9 +10,7 @@ export const initializeGasStore = (chainId: string, providerName: string) =>
       draft.gasPriceStore[chainId] = {};
     }
 
-    if (!draft.gasPriceStore[chainId]![providerName]) {
-      draft.gasPriceStore[chainId]![providerName] = { gasPrices: [], sponsorLastUpdateTimestampMs: {} };
-    }
+    draft.gasPriceStore[chainId]![providerName] = { gasPrices: [], sponsorLastUpdateTimestampMs: {} };
```

Comment:
This will now always to the initialisation for a particular provider, other provider data for the chain is untouched. 

Maybe it could also reset the gas price store for all providers and all chains? (It's not really important because the initialisation is now called once per provider).